src/models/state.ts

```diff
@@ -1,6 +1,24 @@
+import { GraphObject } from '../utils/observer.utils';
+
 // Enum is dismissed so user can define custom additional events (numbers)
 export const GraphObjectState = {
   NONE: 0,
   SELECTED: 1,
   HOVERED: 2,
 };
+
+export interface IGraphObjectStateOptions {
+  isToggle?: boolean;
+  isSingle?: boolean;
+}
```

Comment:
In these cases, when it is options, it would be best not to have optionals here, but where you use it, line 17: `Partial<IGraphObjectStateOptions>` where options can also be optional, e.g. `options?: Partial<...>`.

If you move external logic (having optional params) too deep into the type, it is harder to handle that in some other cases. E.g. if you wish to create default params, you could then just say: `const default: IGraphObjectStateOptions = { ... }` and you would be forced to define both. If you would use `options` from `Parameters` you would see that this object is optional here with all its children. The context of when is something optional, when it isn't is being kept.